src/libecalc/common/time_utils.py

```diff
@@ -53,6 +53,21 @@ def intersects(first: Period, second: Period) -> bool:
         """
         return first.start in second or second.start in first
 
+    @staticmethod
+    def intersection(first: Period, second: Period) -> Period:
+        """
+        Args:
+            first:
+            second:
+
+        Returns:
+
+        """
+        if not Period.intersects(first, second):
+            return Period()
+
+        return Period(max(first.start, second.start), min(first.end, second.end))
```

Comment:
nifty :) I have seen "algorithms" for periods/intervals and intersection logic that stretches over "pages" before :)